### PR TITLE
Adding Microsoft.DevSpaces SDK for 2018-06-01-preview version

### PR DESCRIPTION
**Pull request by sanar-microsoft**
## Description

Link to the rest api spec pull request: https://github.com/Azure/azure-rest-api-specs/pull/3186 

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/DevSpaces/Management.DevSpaces/generate.cmd [hunk 1]
```diff
@@ -0,0 +1,7 @@
+::
+:: Microsoft Azure SDK for Net - Generate library code
+:: Copyright (C) Microsoft Corporation. All Rights Reserved.
+::
+
+@echo off
+call %~dp0..\..\..\..\tools\generate.cmd devspaces/resource-manager %*
```
Comment (dsgouda) [inline src/SDKs/DevSpaces/Management.DevSpaces/generate.cmd]: Please replace this with a generate.ps1 similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Monitor/Management.Monitor/generate.ps1) and regenerate the code.
Please commit any .txt files that get generated.

File: src/SDKs/DevSpaces/Management.DevSpaces/Microsoft.Azure.Management.DevSpaces.csproj [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<Project Sdk="Microsoft.NET.Sdk">
+	<!-- Please do not move/edit code below this line -->
+	<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+	<!-- Please do not move/edit code below this line -->
+
+    <PropertyGroup>
+		<PackageId>Microsoft.Azure.Management.DevSpaces</PackageId>
+		<Description>Provides management operations support for Azure Dev Spaces.</Description>
+		<Version>1.0.0</Version>
```
Comment (dsgouda) [inline src/SDKs/DevSpaces/Management.DevSpaces/Microsoft.Azure.Management.DevSpaces.csproj]: If this is a preview release, please start versioning from `0.9.0-preview`

**Review by dsgouda — CHANGES_REQUESTED** [review]
@sanar-microsoft Please create a test project and add tests related to the sdk, we do not allow sdks without tests

After generating the code, please run `msbuild build.proj /p:Scope=SDKs\DevSpaces` and commit all the changes generated.

File: src/SDKs/DevSpaces/AzSdk.RP.props [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--This file and it's contents are updated at build time moving or editing might result in build failure. Take due deligence while editing this file-->
+  <PropertyGroup>
+    <AzureApiTag />
```
Comment (shahabhijeet) [inline src/SDKs/DevSpaces/AzSdk.RP.props]: @sanar-microsoft missing meta data.
Can you build the generated SDK as
msbuild build.proj /t:Build /p:Scope=SDKs\DevSpaces

File: src/SDKs/DevSpaces/Management.DevSpaces/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+
+[assembly: AssemblyTitle("Microsoft Azure Dev Spaces Management Library")]
+[assembly: AssemblyDescription("Provides management operations support for Azure Dev Spaces.")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/DevSpaces/Management.DevSpaces/Properties/AssemblyInfo.cs]: please start your first release versioning scheme at 0.9.0.0 for your first preview release

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**